File: Workspace_Java/Java05_Switch/src/com/test03/Score.java
package com.test03;

public class Score {
	
	/*
	 * 1. int형 값 3개를 받아서, 3으로 나눠서 평균 값을 리턴하는 getAvg 라는 메소드를 만들자.
	 * 접근은 어디서나 가능하며, non-static으로 만들자
	 */
	public double getAvg(int kor, int eng, int math) {
		int sum = kor + eng + math;
		double res = (double)sum / 3;
		
		return res;
	}
	
	
	
	/*
	 * 2. double형 평균값 하나를 받아서,
	 * 값이 90 ~ 100 사이면 "A"
	 * 80 ~ 89 면 "B"
	 * 70 ~ 79 면 "C"
	 * 60 ~ 69 면 "D"
	 *    ~ 59 면 "F"
	 * 를 리턴하는 getGrade 메소드 만들자.
	 * 어디서나 접근 가능하며, non-static으로 만들자.
	 * 그리고, 기능은 switch로 만들자 (if 사용 금지!)
	 */
	public String getGrade(double avg) {
		int score = (int)avg/10;
		char grade = ' ';
		
		switch (score) {
		case 10:
		case 9:
			grade = 'A';
			break;
		case 8:
			grade = 'B';
			break;
		case 7:
			grade = 'C';
			break;
		case 6:
			grade = 'D';
			break;
		case 5: case 4: case 3: case 2: case 1: case 0:
			grade = 'F';
			break;
		default:
			grade = 'X';
			break;
		}
		
		return String.valueOf(grade);
	}
	
	// 3. MTest의 main 메소드에서 getAvg와 getGrade를 사용하여 홍길동과 이순신의 등급을 출력하자.

}

File: Workspace_Web/Jsp10_Ajax02/WebContent/resources/js/score.js


function getParameterValues() {
	var name = "name=" + encodeURIComponent(document.getElementById("name").value);
	var kor = "kor=" + document.getElementById("kor").value;
	var eng = "eng=" + document.getElementById("eng").value;
	var math = "math=" + document.getElementById("math").value;
	
	return "?" + name + "&" + kor + "&" + eng + "&" + math;
}

function load() {
	var url = "score.do" + getParameterValues();
	
	httpRequest = new XMLHttpRequest();				// server와 통신하는 것을 도와주는 객체
	httpRequest.onreadystatechange = callback;		// 처리할 함수
	httpRequest.open("GET", url, true);				// true : 비동기 / false : 동기
	httpRequest.send();								// get : send / post : send("data")
}

function callback() {
	alert("readystate : " + httpRequest.readyState);
	// readyState == 4 : 통신 완료 되었다면.
	if (httpRequest.readyState == 4) {
		alert("status : " + httpRequest.status);
		// 통신이 성공적으로 완료되었다면.
		if (httpRequest.status == 200) {
			// responseText : 요청 후 응답받은 문자열
			var obj = JSON.parse(httpRequest.responseText);
			document.getElementById("result").innerHTML = decodeURIComponent(obj.name) + "의 총점 : " + obj.sum +
			"\n평균 : " + obj.avg;
		} else {
			alert("통신 실패");
		}
	}

/*
	XMLHttpRequest : javascript object. http를 통한 데이터 송수신 지원

	<onreadystatechange>
	- readystate
		0: unitialized - 실행(load)되지 않음
		1: loading - 로드 중
		2: loaded - 로드 됨
		3: interactive - 통신 됨
		4: complete - 통신 완료

	- status
		200 : 성공
		400 : bad request	
		401 : unauthorized
		403 : forbidden
		404 : not found
		415 : unseupported media tyep
		500 : internal server error
		
		
		******
		encodeURIComponent : 모든 문자를 인코딩하는 함수
		decodeURIComponent : encodeURIComponent()로 인코딩한 문자열을 디코딩하는 함수
		JSON.parse : json형태의 문자열으 json 객체로 변환 (string -> json object)
		JOSN.stringify : javascript 객체(json 형태로 변환할 수 있는)를 json 형태의 문자열로 변환 (object -> json string)
		
	*/	
}